src/mdoc_zk.rs

```diff
@@ -0,0 +1,257 @@
+        pky: String,
+        /// Handoff session binding data.
+        #[serde(deserialize_with = "hex::serde::deserialize")]
+        transcript: Vec<u8>,
+        /// Attributes to be presented.
+        attributes: Vec<TestVectorAttribute>,
+        /// Current time, in RFC 3339 format.
+        now: String,
+        /// Inputs to the signature circuit.
+        #[serde(deserialize_with = "hex::serde::deserialize")]
+        signature_input: Vec<u8>,
+        /// Inputs to the hash circuit.
+        #[serde(deserialize_with = "hex::serde::deserialize")]
+        hash_input: Vec<u8>,
+        /// Verifier's share of MAC key.
+        #[serde(deserialize_with = "hex::serde::deserialize")]
+        mac_verifier_key_share: Vec<u8>,
+        /// Prover's shares of MAC keys.
+        #[serde(deserialize_with = "hex::serde::deserialize")]
+        mac_prover_key_shares: Vec<u8>,
+    }
+
+    /// Presented attribute, as represented in a test vector.
+    #[derive(Deserialize)]
+    struct TestVectorAttribute {
+        id: String,
+        #[serde(deserialize_with = "hex::serde::deserialize")]
+        cbor_value: Vec<u8>,
+    }
+
+    fn load_witness_test_vector() -> WitnessTestVector {
+        serde_json::from_slice(include_bytes!(
+            "../test-vectors/mdoc_zk/witness_test_vector.json"
+        ))
+        .unwrap()
+    }
+
+    #[ignore = "failing, witness preparation is incomplete"]
+    #[wasm_bindgen_test(unsupported = test)]
+    fn witness_preparation() {
+        let test_vector = load_witness_test_vector();
+
+        let mut issuer_pkx = hex::decode(&test_vector.pkx[2..]).unwrap();
+        let mut issuer_pky = hex::decode(&test_vector.pky[2..]).unwrap();
+        // Switch from big endian to little endian before decoding field elements.
+        issuer_pkx.reverse();
+        issuer_pky.reverse();
+        let issuer_pkx = FieldP256::try_from(issuer_pkx.as_slice()).unwrap();
+        let issuer_pky = FieldP256::try_from(issuer_pky.as_slice()).unwrap();
+
+        let attributes = test_vector
+            .attributes
+            .iter()
+            .map(|attr| Attribute {
+                id: attr.id.clone(),
+                cbor_value: attr.cbor_value.clone(),
+            })
+            .collect::<Vec<_>>();
+
+        let mac_verifier_key_share =
+            Field2_128::try_from(test_vector.mac_verifier_key_share.as_slice()).unwrap();
+        let mac_prover_key_shares = test_vector
+            .mac_prover_key_shares
+            .chunks_exact(Field2_128::num_bytes())
+            .map(|bytes| Field2_128::try_from(bytes).unwrap())
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap();
+
+        let mut inputs = CircuitInputs::new(
+            CircuitVersion::V6,
+            &test_vector.mdoc,
+            [issuer_pkx, issuer_pky],
+            &test_vector.transcript,
+            &attributes,
+            &test_vector.now,
+            &[Field2_128::ZERO; 6],
```

Comment:
Why six zeroes?

Reply:
This is just a placeholder for now, we'll want to calculate the real values later. I added a comment to that effect. I also realized I swapped two of the arguments here!